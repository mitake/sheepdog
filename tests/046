#!/bin/bash

# Test vdi deletion

seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1        # failure is the default!

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_cleanup

for i in 0 1 2; do
    _start_sheep $i
done

_wait_for_sheep 3

$COLLIE cluster format -c 2

echo "delete snapshot image before current one"
$COLLIE vdi create test 4G
$COLLIE vdi snapshot test -s snap
$COLLIE vdi list | _filter_short_date

$COLLIE vdi delete test -s snap
$COLLIE vdi delete test -s snap # error
$COLLIE vdi list | _filter_short_date

$COLLIE vdi delete test
$COLLIE vdi delete test # error
$COLLIE vdi list | _filter_short_date

echo "delete current image before snapshot one"
$COLLIE vdi create test 4G
$COLLIE vdi snapshot test -s snap
$COLLIE vdi list | _filter_short_date

$COLLIE vdi delete test
$COLLIE vdi delete test # error
$COLLIE vdi list | _filter_short_date

$COLLIE vdi delete test -s snap
$COLLIE vdi delete test -s snap # error
$COLLIE vdi list | _filter_short_date
