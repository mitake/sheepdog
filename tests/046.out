QA output created by 046
using backend farm store
delete snapshot image before current one
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
s test         1  4.0 GB  0.0 MB  0.0 MB DATE   7c2b25     2          snap
  test         2  4.0 GB  0.0 MB  0.0 MB DATE   7c2b26     2              
Failed to delete test: Failed to find requested tag
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         2  4.0 GB  0.0 MB  0.0 MB DATE   7c2b26     2              
Failed to delete test: No VDI found
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
delete current image before snapshot one
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
s test         1  4.0 GB  0.0 MB  0.0 MB DATE   7c2b25     2          snap
  test         2  4.0 GB  0.0 MB  0.0 MB DATE   7c2b27     2              
Failed to delete test: No VDI found
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
s test         1  4.0 GB  0.0 MB  0.0 MB DATE   7c2b25     2          snap
Failed to delete test: No VDI found
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
